app/controllers/api/tracks/solutions_for_mentoring_controller.rb

```diff
@@ -0,0 +1,16 @@
+module API
+  module Tracks
+    class SolutionsForMentoringController < BaseController
+      def index
+        track = Track.find_by!(slug: params[:track_slug])
```

Comment:
This should be `track = Track.find(params[:track_slug])` in order for FriendlyId history to work I think.